contracts/staking/query.go

```diff
@@ -0,0 +1,73 @@
+package staking
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/0xPolygon/polygon-sdk/contracts/abis"
+	"github.com/0xPolygon/polygon-sdk/state/runtime"
+	"github.com/0xPolygon/polygon-sdk/types"
+	"github.com/umbracle/go-web3"
+	"github.com/umbracle/go-web3/abi"
+)
+
+var (
+	// staking contract address
+	AddrStakingContract = types.StringToAddress("1001")
+
+	// Gas limit used when querying the validator set
+	queryGasLimit uint64 = 100000
+)
+
+func DecodeValidators(method *abi.Method, returnValue []byte) ([]types.Address, error) {
+	decodedResults, err := method.Outputs.Decode(returnValue)
+	if err != nil {
+		return nil, err
+	}
+	results, ok := decodedResults.(map[string]interface{})
+	if !ok {
+		return nil, errors.New("failed type assertion from decodedResults to map")
+	}
+	web3Addresses, ok := results["0"].([]web3.Address)
+	if !ok {
+		return nil, errors.New("failed type assertion from results[0] to []web3.Address")
+	}
+
+	addresses := make([]types.Address, len(web3Addresses))
+	for idx, waddr := range web3Addresses {
+		addresses[idx] = types.Address(waddr)
+	}
+
+	return addresses, nil
+}
+
+type TxQueryHandler interface {
+	Apply(*types.Transaction) (*runtime.ExecutionResult, error)
+	GetNonce(types.Address) uint64
+}
+
+func QueryValidators(t TxQueryHandler, from types.Address) ([]types.Address, error) {
+	method, ok := abis.StakingABI.Methods["validators"]
+	if !ok {
+		return nil, errors.New("validators method doesn't exist in Staking contract ABI")
+	}
+
+	selector := method.ID()
+	res, err := t.Apply(&types.Transaction{
+		From:     from,
+		To:       &AddrStakingContract,
+		Value:    big.NewInt(0),
+		Input:    selector,
+		GasPrice: big.NewInt(0),
```

Comment:
I think it's completely fine to leave it to 0, since there isn't any validator for it in `transition.Apply`, apart from the regular check for `gasPrice * gasLimit`. We've tested this out in non-zero gas networks, and it works fine 👍 

cc @Kourin1996

Reply:
@Vuksan @zivkovicmilos Yes, this transaction is for querying the contract about validators and it will be called directly after the node added the block where the height is the end of an epoch and not be applied to the state. For that reason, it doesn't affect to the order and account balance even though we set higher gas price.